src/test/rgw/lua/test_lua.py

```diff
@@ -0,0 +1,425 @@
+import logging
+import json
+import tempfile
+import random
+import threading
+import subprocess
+import socket
+import time
+import os
+import string
+import boto
+from botocore.exceptions import ClientError
+from http import server as http_server
+from random import randint
+import hashlib
+from nose.plugins.attrib import attr
+import boto3
+import datetime
+from dateutil import parser
+
+from boto.s3.connection import S3Connection
+
+from . import(
+    get_config_host,
+    get_config_port,
+    get_access_key,
+    get_secret_key
+    )
+
+from .api import delete_all_objects, \
+    admin
+
+from nose import SkipTest
+from nose.tools import assert_not_equal, assert_equal, assert_in
+import boto.s3.tagging
+
+# configure logging for the tests module
+log = logging.getLogger(__name__)
+
+
+num_buckets = 0
+run_prefix=''.join(random.choice(string.ascii_lowercase) for _ in range(6))
+
+def gen_bucket_name():
+    global num_buckets
+
+    num_buckets += 1
+    return run_prefix + '-' + str(num_buckets)
+
+
+def set_contents_from_string(key, content):
+    try:
+        key.set_contents_from_string(content)
+    except Exception as e:
+        print('Error: ' + str(e))
+
+
+def get_ip():
+    return 'localhost'
+
+
+def get_ip_http():
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        # address should not be reachable
+        s.connect(('10.255.255.255', 1))
+        ip = s.getsockname()[0]
+    finally:
+        s.close()
+    return ip
+
+
+def connection():
+    hostname = get_config_host()
+    port_no = get_config_port()
+    vstart_access_key = get_access_key()
+    vstart_secret_key = get_secret_key()
+
+    conn = S3Connection(aws_access_key_id=vstart_access_key,
+                  aws_secret_access_key=vstart_secret_key,
+                      is_secure=False, port=port_no, host=hostname, 
+                      calling_format='boto.s3.connection.OrdinaryCallingFormat')
+
+    return conn
+
+
+def connection2():
+    hostname = get_config_host()
+    port_no = 8001
+    vstart_access_key = get_access_key()
+    vstart_secret_key = get_secret_key()
+
+    conn = S3Connection(aws_access_key_id=vstart_access_key,
+                  aws_secret_access_key=vstart_secret_key,
+                      is_secure=False, port=port_no, host=hostname, 
+                      calling_format='boto.s3.connection.OrdinaryCallingFormat')
+
+    return conn
+
+
+def connection3():
+    hostname = get_config_host()
+    port_no = get_config_port()
+    vstart_access_key = get_access_key()
+    vstart_secret_key = get_secret_key()
+
+    conn = boto3.client(service_name="s3",
+            aws_access_key_id=vstart_access_key,
+            aws_secret_access_key=vstart_secret_key,
+            endpoint_url="http://"+hostname+":"+str(port_no))
+
+    return conn
+
+
+def connection2():
```

Comment:
This function is repeated https://github.com/ceph/ceph/pull/52931/files#diff-59cc159382fd89820bec962c99bbaf49849c1990e5da862e6f560bfd9c399c9cR87